src-java/rule-manager/rule-manager-implementation/src/main/java/org/openkilda/rulemanager/adapter/PersistenceDataAdapter.java

```diff
@@ -0,0 +1,124 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.rulemanager.adapter;
+
+import org.openkilda.model.Flow;
+import org.openkilda.model.FlowEncapsulationType;
+import org.openkilda.model.FlowPath;
+import org.openkilda.model.FlowTransitEncapsulation;
+import org.openkilda.model.PathId;
+import org.openkilda.model.Switch;
+import org.openkilda.model.SwitchId;
+import org.openkilda.model.SwitchProperties;
+import org.openkilda.model.TransitVlan;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.repositories.FlowPathRepository;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+import org.openkilda.persistence.repositories.SwitchPropertiesRepository;
+import org.openkilda.persistence.repositories.SwitchRepository;
+import org.openkilda.persistence.repositories.TransitVlanRepository;
+import org.openkilda.persistence.repositories.VxlanRepository;
+import org.openkilda.rulemanager.DataAdapter;
+
+import lombok.Builder;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class PersistenceDataAdapter implements DataAdapter {
+
+    private final FlowPathRepository flowPathRepository;
+    private final SwitchRepository switchRepository;
+    private final SwitchPropertiesRepository switchPropertiesRepository;
+    private final TransitVlanRepository transitVlanRepository;
+    private final VxlanRepository vxlanRepository;
+
+    private final Set<PathId> pathIds;
+    private final Set<SwitchId> switchIds;
+
+    private Map<PathId, Flow> flowCache;
```

Comment:
What is the purpose of these caches? Even changes within the same persistence context (transaction) wont invalidate them.

Reply:
PersistenceDataAdapter class is designed to be used for one operation only. So consequent calls to it's API should return the same result and not producing a lot of DB calls. No invalidation is designed.